pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -99,8 +99,9 @@ stdenv.mkDerivation rec {
 
     # Fix xkeyboard config path for Qt
     for path in mathematica Mathematica; do
-      line=$(grep -n QT_PLUGIN_PATH $path | sed 's/:.*//')
-      sed -i -e "$line iexport QT_XKB_CONFIG_ROOT=\"${xkeyboard_config}/share/X11/xkb\"" $path
+      for line in $(grep -n QT_PLUGIN_PATH $path | sed 's/:.*//'); do
```

Comment:
What kind of lines do we get from this "grep -n QT_PLUGIN_PATH $path"?

Reply:
IIRC the purpose of this change is to setup QT properly to use Mathematica in KDE. It exports `QT_XKB_CONFIG_ROOT`, such as in the following case (from `$out/libexec/Mathematica/Executables/mathematica`):

```
export QT_XKB_CONFIG_ROOT="/nix/store/17588p39scnw7y71y7f2wvhm5mpa63wi-xkeyboard-config-2.24/share/X11/xkb"
export QT_PLUGIN_PATH=${M_LIBRARY_PATH}/Qt-Plugins
```

The first line is added by this change.

Reply:
@herberteuler 
Why not just do:
```bash
sed -i -e "/export QT_PLUGIN_PATH/iexport QT_XKB_CONFIG_ROOT=\"${xkeyboard_config}/share/X11/xkb\"" $path
```
?

Also do you think this new change is reasonable?

Reply:
Using `sed`'s built-in function to reference places to change is indeed better than referencing with line numbers, especially in this case, because after each iteration of the `for` loop the old line numbers will become invalid.

Considering that exporting `QT_XKB_CONFIG_ROOT` needs to be done only once, this is perhaps better:

```
sed -i -e "/^\"\${MathematicaFE/iexport QT_XKB_CONFIG_ROOT=\"${xkeyboard_config}/share/X11/xkb\"\n" $path
```

As for dropping 11.3, I think perhaps it would be better to still provide it in another file, e.g. `11.nix`, just like what we do for 10 and 9 now.

Reply:
@herberteuler Could you provide a PR for that? I guess, we might as well insert this export on the line 2 instead of matching by the regex.

I don't mind keeping 11.3 if someone wants to use it and willing to maintain.

Reply:
Also thanks for pointing out that we have Mathematica 9 and 10. I didn't know.